consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -75,13 +75,16 @@ impl PersistentSafetyStorage {
         // storage.
         if let Err(diem_secure_storage::Error::KeyAlreadyExists(_)) = result {
             warn!("Attempted to re-initialize existing storage");
+            let waypoint_version: u64 = internal_store.get(WAYPOINT).map(|v| v.value)?;
```

Comment:
Hmm, do you want to error out if the waypoint version is wrong?  Since this now provides a new condition that might cause it to exit with an `Err()` (this probably shouldn't happen). 

Oh, but nevermind, your exit condition is that internal_store is set for `WAYPOINT`.

Reply:
I removed setting/updating waypoint counters here.